com.unity.render-pipelines.universal/Editor/ShaderGraph/Includes/DepthOnlyPass.hlsl

```diff
@@ -23,7 +23,15 @@ half4 frag(PackedVaryings packedInput) : SV_TARGET
         clip(surfaceDescription.Alpha - surfaceDescription.AlphaClipThreshold);
     #endif
 
-    return 0;
+    half4 outColor = 0;
+    #ifdef SCENESELECTIONPASS
```

Comment:
I suppose this makes it technically not a "DepthOnly" shader, which now becomes somewhat confusing name.
I'm not sure if coupling these together is good idea.
Is this how it's done for other pipes too?

Also can this combine with depthNormalsOnly pass?

Reply:
This is currently how HDRP does it (and now built-in because I've already put this in there). I agree it could be in a separate pass file if we desire.

Reply:
Oh and to be clear, this is a separate pass being run, it's just re-using the depth-only hlsl file. I assume that's because this is the smallest pass file that runs the vertex description function and checks for alpha clipping and doesn't do lighting.

Reply:
I've moved them to a separate file and made them shared between the 3d and 2d passes